src/graphics/vertex_buffer.rs

```diff
@@ -30,6 +31,12 @@ pub struct VertexBuffer {
     vertex_buffer: NonNull<sfVertexBuffer>,
 }
 
+#[derive(Debug, Error)]
+pub enum VertexBufferError {
+    #[error("`{0}`")]
+    Message(String),
```

Comment:
Holding a String is a bit heavy here, since this is just two statically determinable errors.
I would personally create two types here.
```Rust
#[derive(Debug, Error)]
#[error("failed to create vertex buffer")]
pub struct VertexBufferCreateError;

#[derive(Debug, Error)]
#[error("failed to clone vertex buffer")]
pub struct VertexBufferCloneError;
```